SalesforcePluginApp.ts

```diff
@@ -0,0 +1,831 @@
+        const contentParsed = JSON.parse(contentStringified);
+
+        console.log(
+          'Find Session Variables from Persistent Storage, Response: ',
+        );
+        console.log(contentParsed);
+
+        persisantAffinity = contentParsed.affinityToken;
+        persistantKey = contentParsed.key;
+      } catch {
+        console.log(
+          'Error: Cannot Find Session Variables from Persistent Storage.',
+        );
+      }
+
+      if (
+        message.text === 'Closed by visitor' &&
+        persisantAffinity &&
+        persistantKey
+      ) {
+        const closeLiveAgentChatHttpRequest: IHttpRequest = {
+          headers: {
+            'X-LIVEAGENT-API-VERSION': '49',
+            'X-LIVEAGENT-AFFINITY': persisantAffinity,
+            'X-LIVEAGENT-SESSION-KEY': persistantKey,
+          },
+          data: {
+            reason: 'client',
+          },
+        };
+
+        http
+          .post(
+            `${salesforceChatApiEndpoint}Chasitor/ChatEnd`,
+            closeLiveAgentChatHttpRequest,
+          )
+          .then(async (res) => {
+            console.log('Closing Liveagent Chat, Response:');
+            await persistence.removeByAssociation(assoc);
+            console.log(res);
+          })
+          .catch((error) => {
+            console.log('Closing Liveagent Chat, Error:');
+            console.log(error);
+          });
+      }
+
+      if (
+        message.sender.username !== salesforceBotUsername &&
+        message.text !== 'Closed by visitor' &&
+        persisantAffinity &&
+        persistantKey
+      ) {
+        const sendMessageToLiveAgentHttpRequest: IHttpRequest = {
+          headers: {
+            'X-LIVEAGENT-API-VERSION': '49',
+            'X-LIVEAGENT-AFFINITY': persisantAffinity,
+            'X-LIVEAGENT-SESSION-KEY': persistantKey,
+          },
+          data: {
+            text: message.text,
+          },
+        };
+
+        http
+          .post(
+            `${salesforceChatApiEndpoint}Chasitor/ChatMessage`,
+            sendMessageToLiveAgentHttpRequest,
+          )
+          .then((res) => {
+            console.log('Sending Message To Liveagent, Response:');
+            console.log(res);
+          })
+          .catch((error) => {
+            console.log('Sending Message To Liveagent, Error:');
+            console.log(error);
+          });
+      }
+
+      if (persisantAffinity && persistantKey) {
+        const pullingMesssagesSFAHttpRequest: IHttpRequest = {
+          headers: {
+            'X-LIVEAGENT-API-VERSION': '49',
+            'X-LIVEAGENT-AFFINITY': persisantAffinity,
+            'X-LIVEAGENT-SESSION-KEY': persistantKey,
+          },
+        };
+
+        http
+          .get(
+            `${salesforceChatApiEndpoint}System/Messages`,
+            pullingMesssagesSFAHttpRequest,
+          )
+          .then(async (response) => {
+            const { content } = response;
```

Comment:
When parsing the content JSON, it errors with this after the user sends a message to the agent (i added the response received log point for debugging myself)
```
I20200625-12:51:08.868(-4)? RESPONSE RECIEVED:
I20200625-12:51:08.869(-4)? {
I20200625-12:51:08.869(-4)?   statusCode: 409,
I20200625-12:51:08.869(-4)?   content: 'Duplicate Message Observer',
I20200625-12:51:08.869(-4)?   headers: {
I20200625-12:51:08.869(-4)?     connection: 'close',
I20200625-12:51:08.869(-4)?     'content-length': '26',
I20200625-12:51:08.870(-4)?     'content-type': 'application/json',
I20200625-12:51:08.870(-4)?     'access-control-allow-origin': '*',
I20200625-12:51:08.870(-4)?     'access-control-allow-credentials': 'true',
I20200625-12:51:08.870(-4)?     'cache-control': 'no-cache',
I20200625-12:51:08.870(-4)?     pragma: 'no-cache',
I20200625-12:51:08.870(-4)?     expires: '-1'
I20200625-12:51:08.870(-4)?   },
I20200625-12:51:08.870(-4)?   data: null
I20200625-12:51:08.870(-4)? }
I20200625-12:51:08.870(-4)? Pulling Messages from Liveagent, Error:
I20200625-12:51:08.871(-4)? SyntaxError: Unexpected token D in JSON at position 0
I20200625-12:51:08.871(-4)?     at JSON.parse (<anonymous>)
I20200625-12:51:08.871(-4)?     at evalmachine.<anonymous>:424:48
I20200625-12:51:08.871(-4)?     at /Users/dlassalle/.meteor/packages/promise/.0.11.2.1e1wn8z.joczg++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/meteor-promise/fiber_pool.js:43:40
```

this may explain it 
https://salesforce.stackexchange.com/questions/245558/duplicate-message-observer-on-pulling-messages-from-live-agent-server

Reply:
Not exactly sure, since on my end this error is not showing up. Maybe I will try doing a new installation and see if this error shows up 👌

Reply:
So I tried 3 different versions of Rocket Chat, but none of them has this issue coming up.

Reply:
hmm, maybe its because we have onmichannel enabled in salesforce 🤔 , but its not a huge issue at the moment - I have other problems I'm working through :)